llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -737,20 +722,35 @@ processInputModule(std::unique_ptr<Module> M) {
       for (size_t i = 0; i != MMsWithDefaultSpecConsts.size(); ++i) {
         module_split::ModuleDesc &IrMD = MMsWithDefaultSpecConsts[i];
         IsBF16DeviceLibUsed |= isSYCLDeviceLibBF16Used(IrMD.getModule());
-        saveModule(Tables, IrMD, ID, OutIRFileName);
+        saveModule(Tables, IrMD, ID, OutputPrefix, OutIRFileName);
       }
 
       ++ID;
     }
   }
 
   if (IsBF16DeviceLibUsed && (DeviceLibDir.getNumOccurrences() > 0)) {
-    saveDeviceLibModule(Tables, ID, "libsycl-fallback-bfloat16.bc");
-    saveDeviceLibModule(Tables, ID + 1, "libsycl-native-bfloat16.bc");
+    saveDeviceLibModule(Tables, OutputPrefix, ID,
```

Comment:
Maybe it is cleaner to populated a global constant with the list of device library module filenames and use that here?